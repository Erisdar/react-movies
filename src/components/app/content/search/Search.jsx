import React from 'react';
import Button from 'react-bootstrap/Button';
import Form from 'react-bootstrap/Form';
import FindMovieText from './findMovieText/FindMovieText';
import './search.css';

export default ({ searchType, searchValue, changeSearchValue, search, setSearchTypeTitle, setSearchTypeGenre }) => {

    let onSubmit = event => {
        event.preventDefault();
        search({ searchType, searchValue });
    }

    return (
        <form className="search content__search" onSubmit={onSubmit}>
            <Form.Group controlId="search__film-input">
                <Form.Label>
                    <FindMovieText />
                </Form.Label>
                <Form.Control type="text" maxLength="50" value={searchValue} className="search__film-input" data-cy='input'
                    placeholder={'Enter film ' + searchType} onChange={event => changeSearchValue(event.target.value)} />
                <div className="d-flex justify-content-center align-items-center search__button-group">
                    <div>SEARCH BY</div>
                    <Button type="button" variant={searchType == "title" ? "danger" : "secondary"} data-cy="title" onClick={setSearchTypeTitle}
                        className="search__button-title">TITLE</Button>
                    <Button type="button" variant={searchType == "genres" ? "danger" : "secondary"} data-cy="genre" onClick={setSearchTypeGenre}
                        className="search__button-genres">GENRE</Button>
                    <Button type="submit" data-cy="search" variant="danger" size="lg" className="search__button-search ml-auto">SEARCH</Button>
                </div>
            </Form.Group>
        </form>
    );
};


