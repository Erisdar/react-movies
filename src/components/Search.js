import React, { Component } from 'react';
import FormControl from 'react-bootstrap/FormControl';
import InputGroup from 'react-bootstrap/InputGroup';
import '../styles/search.css';
import FindMovieText from './FindMovieText';
import SearchButtonGroup from './SearchButtonGroup';

export default class Search extends Component {
    constructor(props) {
        super(props);
        this.state = {
            searchType: 'title',
            searchValue: ''
        }
    };

    setSearchValue(value) {
        this.setState({
            searchValue: value
        })
    };

    setSearchType(type) {
        this.setState({
            searchType: type
        })
    };

    render() {
        return (
            <div className="search">
                <FindMovieText />
                <InputGroup className="mb-15">
                    <FormControl onInput={e => this.setSearchValue(e.target.value)} placeholder="Film" aria-label="Film" aria-describedby="basic-addon1" />
                </InputGroup>
                <SearchButtonGroup searchType={this.state.searchType} changeSearchType={(type) => this.setSearchType(type)}
                    search={() => this.props.onSearch(this.state.searchType, this.state.searchValue)} />
            </div>
        )
    }
};


