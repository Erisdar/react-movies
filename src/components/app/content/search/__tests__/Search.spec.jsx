import { shallow, mount } from 'enzyme';
import React from 'react';
import Search from '../Search';

describe('Search component', () => {
    it('should be render correctly', () => {
        const component = shallow(<Search />);
        expect(component).toMatchSnapshot();
    });

    it('should call search() onSumbit', () => {
        const search = jest.fn();
        const component = mount(<Search search={search} />);
        component.find('button.search__button-search').simulate('submit');
        expect(search).toHaveBeenCalled();
    });

    it('should change searchValue onInput', () => {
        const changeSearchValue = jest.fn();
        const component = shallow(<Search changeSearchValue={changeSearchValue} />);
        component.find('.search__film-input').simulate('change', { target: { value: 'Hello' } });
        expect(changeSearchValue).toHaveBeenCalledWith('Hello');
    });
});