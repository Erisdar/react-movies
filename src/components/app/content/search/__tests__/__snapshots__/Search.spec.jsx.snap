// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component should be render correctly 1`] = `
<form
  className="search content__search"
  onSubmit={[Function]}
>
  <FormGroup
    as="div"
    controlId="search__film-input"
  >
    <FormLabel
      column={false}
      srOnly={false}
    >
      <_default />
    </FormLabel>
    <FormControl
      as="input"
      className="search__film-input"
      data-cy="input"
      maxLength="50"
      onChange={[Function]}
      placeholder="Enter film undefined"
      type="text"
    />
    <div
      className="d-flex justify-content-center align-items-center search__button-group"
    >
      <div>
        SEARCH BY
      </div>
      <Button
        active={false}
        className="search__button-title"
        data-cy="title"
        disabled={false}
        type="button"
        variant="secondary"
      >
        TITLE
      </Button>
      <Button
        active={false}
        className="search__button-genres"
        data-cy="genre"
        disabled={false}
        type="button"
        variant="secondary"
      >
        GENRE
      </Button>
      <Button
        active={false}
        className="search__button-search ml-auto"
        data-cy="search"
        disabled={false}
        size="lg"
        type="submit"
        variant="danger"
      >
        SEARCH
      </Button>
    </div>
  </FormGroup>
</form>
`;
